test/fbamodel_importer_test.py

```diff
@@ -0,0 +1,174 @@
+        config_file = environ.get('KB_DEPLOYMENT_CONFIG', None)
+        cls.cfg = {}
+        config = ConfigParser()
+        config.read(config_file)
+        for nameval in config.items('kb_uploadmethods'):
+            cls.cfg[nameval[0]] = nameval[1]
+        authServiceUrl = cls.cfg.get('auth-service-url',
+                                     "https://kbase.us/services/authorization/Sessions/Login")
+        auth_client = _KBaseAuth(authServiceUrl)
+        cls.user_id = auth_client.get_user(cls.token)
+        # WARNING: don't call any logging methods on the context object,
+        # it'll result in a NoneType error
+        cls.ctx = MethodContext(None)
+        cls.ctx.update({'token': cls.token,
+                        'user_id': cls.user_id,
+                        'provenance': [
+                            {'service': 'kb_uploadmethods',
+                             'method': 'please_never_use_it_in_production',
+                             'method_params': []
+                             }],
+                        'authenticated': 1})
+        cls.wsURL = cls.cfg['workspace-url']
+        cls.wsClient = workspaceService(cls.wsURL, token=cls.token)
+        cls.serviceImpl = kb_uploadmethods(cls.cfg)
+        cls.dfu = DataFileUtil(os.environ['SDK_CALLBACK_URL'], token=cls.token)
+        cls.scratch = cls.cfg['scratch']
+        cls.shockURL = cls.cfg['shock-url']
+
+    @classmethod
+    def tearDownClass(cls):
+        if hasattr(cls, 'wsName'):
+            cls.wsClient.delete_workspace({'workspace': cls.wsName})
+            print('Test workspace was deleted')
+
+    @classmethod
+    def delete_shock_node(cls, node_id):
+        header = {'Authorization': 'Oauth {0}'.format(cls.token)}
+        requests.delete(cls.shockURL + '/node/' + node_id, headers=header,
+                        allow_redirects=True)
+        print('Deleted shock node ' + node_id)
+
+    def getWsClient(self):
+        return self.__class__.wsClient
+
+    def getWsName(self):
+        if hasattr(self.__class__, 'wsName'):
+            return self.__class__.wsName
+        suffix = int(time.time() * 1000)
+        wsName = "test_kb_uploadmethods_" + str(suffix)
+        ret = self.getWsClient().create_workspace({'workspace': wsName})  # noqa
+        self.__class__.wsName = wsName
+        return wsName
+
+    def getImpl(self):
+        return self.__class__.serviceImpl
+
+    def getContext(self):
+        return self.__class__.ctx
+
+    def mock_download_staging_file(params):
+        print 'Mocking DataFileUtilClient.download_staging_file'
+        print params
+
+        fq_filename = params.get('staging_file_subdir_path')
+        fq_path = os.path.join('/kb/module/work/tmp', fq_filename)
+        shutil.copy(os.path.join("data", fq_filename), fq_path)
+
+        return {'copy_file_path': fq_path}
+
+    @patch.object(DataFileUtil, "download_staging_file", side_effect=mock_download_staging_file)
```

Comment:
remove this? I think you are not using it with this test.

Reply:
Actually it was because I tested if compounds_file was set in tsv case after downloading the files. I just moved this test up so it fails sooner